src/mmw/apps/modeling/mapshed.py

```diff
@@ -24,3 +34,70 @@ def day_lengths(geom):
                                                       (month * 30.4375 - 5)))
 
     return lengths
+
+
+def nearest_weather_stations(geom, n=NUM_WEATHER_STATIONS):
+    """
+    Given a geometry, returns a list of the n closest weather stations to it
+    """
+    sql = '''
+          SELECT station, location, meanrh, meanwind, meanprecip,
+                 begyear, endyear, eroscoeff, rain_cool, rain_warm,
+                 etadj, grw_start, grw_end
+          FROM ms_weather_station
+          ORDER BY geom <-> ST_SetSRID(ST_GeomFromText(%s), 4326)
+          LIMIT %s;
+          '''
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql, [geom.wkt, n])
+
+        # Return all rows from cursor as namedtuple
+        weather_station = namedtuple('WeatherStation',
+                                     [col[0] for col in cursor.description])
+        return [weather_station(*row) for row in cursor.fetchall()]
+
+
+def growing_season(ws):
+    """
+    Given an array of weather stations, returns an array of 12 integers, each 1
+    or 0, indicating whether it is a growing season or not respectively.
+    We adopt a liberal approach, unioning the ranges to get a superset which is
+    a growing season for any weather station.
+    """
+
+    start = min([MONTHS.index(w.grw_start) for w in ws])
+    end = max([MONTHS.index(w.grw_end) for w in ws])
```

Comment:
This may be a question for BME, but does it make sense to use the average here? Or are the weather stations close enough where it doesn't really matter?